blueocean-dashboard/src/main/js/components/RunDetailsHeader.jsx

```diff
@@ -119,12 +119,13 @@ class RunDetailsHeader extends Component {
             </div>
         );
 
-        const commitIdString = run.commitId || '—';
+        const commitIdString = changeSet.length && changeSet[changeSet.length - 1].commitId || '—';
```

Comment:
@NicuPascu check but I think the `CommitId` component will handle converting a null commitId to `–` for you.